File: Python/GnomeSort.py
# Python program to implement Gnome Sort
from typing import List, TypeVar

T = TypeVar('T')  # Create a generic type of T


# A function to sort the given list using Gnome sort
def gnome_sort(arr: List[T]) -> List[T]:
    index: int = 0
    n: int = len(arr)
    while index < n:
        if index == 0:
            index = index + 1
        if arr[index] >= arr[index - 1]:
            index += 1
        else:
            arr[index], arr[index - 1] = arr[index - 1], arr[index]
            index -= 1

    return arr


# Driver Code
if __name__ == '__main__':
    arr = [34, 2, 10, -9]

    arr = gnome_sort(arr)
    print("Sorted sequence after applying Gnome Sort :", arr)

File: Go/GnomeSort.go
package main

import "fmt"

func sort(arr []int, n int) {
	for i:=1; i<n-1; i++ {
		if(arr[i] < arr[i-1]) {
			arr[i], arr[i-1] = arr[i-1], arr[i]

			i--
		}
	}
}

func main() {
	var n int

	fmt.Printf("Enter the number of elements: ")
	fmt.Scan(&n)
	arr := make([]int, n)

	fmt.Printf("Enter the %d elements: ", n)
	for i:=0; i<n; i++ {
		fmt.Scan(&arr[i])
	}

	sort(arr, n)
	fmt.Println("Sorted Array:")
	for _, x := range arr {
		fmt.Printf("%d ", x)
	}
	fmt.Println()
}